### ISSUE TITLE
Can Id field mapping be customised?

### ISSUE THREAD

**Issue by bertramn** [bguerout/jongo#239]
I am trying to use a single ObjectMapper configuration to serialise an object to mongodb but also outwards via REST. Unfortunately I am unable to specify a different name for the id field because the Jongo `JacksonIdFieldSelector` is all hard wired to only cater for `_id`.

I used to use mongojack, which understood that the field annotated with `javax.persistence.Id` was meant to be called by the bean property name during serialisation and when it went to the database, it actually translated the field name to `_id`.

Below is what I am trying to achieve. Lets say we have something as simple as this:

```java
import org.jongo.marshall.jackson.oid.Id;
import org.jongo.marshall.jackson.oid.ObjectId;

public class Userr {

	@Id
	@ObjectId
	private String id;

	private String name;

	public String getId() {
		return id;
	}

	public void setId(String id) {
		this.id = id;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

}
```

Serialised/Deserialised when its not coming from or going to mongodb:

```json
{
  "id": "53774d0e0547674f0b001133",
  "name": "fred"
}
```

Serialised to mongodb:

```json
{
  "_id": "53774d0e0547674f0b001133",
  "name": "fred"
}
```

Maybe instead of just mixing in `JsonProperty`, have field name property on `org.jongo.marshall.jackson.oid.Id` that defaults to `_id` and some processing to recognise that the field is the mongo OID but still treat it by the name given (or have a flag to force bean property name handling and only send `_id` to mongo). Could that work? Cheers Niels

**Comment by bguerout** [comment]
This provider shows how to deactivate @Id for an external Jackson : https://github.com/yamsellem/jongo-jersey/blob/master/src/main/java/org/jongo/jersey/provider/JacksonMapperProvider.java

**Comment by bguerout** [comment]
Here is more informations about this topic : https://github.com/bguerout/jongo/issues/212

**Comment by bertramn** [comment]
Thanks Benoit, I did come to a conclusion but it took quite a bit of research to get my case working as I am also using jaxb annotation introspector and could not quite do what jongo-jersey did. Another issue that I ran into was that the jackson ObjectMapper is not easily configurable from a OSGi blueprint application (similar to Spring DI). Now I have a ObjectMapperFactory with a few bits and bolts that pair the annotation introspectors. I'll see if I can merge them into jongo if you would like to have a look and see if this is something you want to include into jongo for others to use out of the box. Kind Regards, Niels
